test/integration/load/load_test.go

```diff
@@ -171,9 +178,94 @@ func TestScaleDeployment(t *testing.T) {
 	if testConfig.Cleanup {
 		err = kubernetes.MustDeleteDeployment(ctx, deploymentsClient, deployment)
 		require.NoError(t, err, "error deleteing load deployment")
+		err = kubernetes.WaitForPodsDelete(ctx, clientset, namespace, podLabelSelector)
+		require.NoError(t, err, "error waiting for pods to delete")
 	}
 }
 
+func TestValidCNSStateDuringScaleAndCNSRestart(t *testing.T) {
+	clientset, err := kubernetes.MustGetClientset()
+	require.NoError(t, err)
+
+	config := kubernetes.MustGetRestConfig(t)
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
+	defer cancel()
+
+	validator, err := validate.CreateValidator(ctx, clientset, config, namespace, testConfig.CNIType, testConfig.RestartCase, testConfig.OSType)
+	require.NoError(t, err)
+
+	err = validator.Validate(ctx)
+	require.NoError(t, err)
+
+	deployment, err := kubernetes.MustParseDeployment(noopDeploymentMap[testConfig.OSType])
+	require.NoError(t, err)
+	deploymentsClient := clientset.AppsV1().Deployments(namespace)
+
+	// Create a deployment
+	err = kubernetes.MustCreateDeployment(ctx, deploymentsClient, deployment)
+	require.NoError(t, err)
+
+	// Scale it up and "skipWait", so CNS restart can happen immediately after scale call is made (while pods are still creating)
+	skipWait := true
+	err = kubernetes.MustScaleDeployment(ctx, deploymentsClient, deployment, clientset, namespace, podLabelSelector, testConfig.ScaleUpReplicas, skipWait)
+	require.NoError(t, err)
+
+	// restart linux CNS (linux, windows)
+	err = kubernetes.RestartCNSDaemonset(ctx, clientset)
```

Comment:
can add  `WaitForPodDaemonset()` to ensure a race condition does not happen after restart.

Reply:
We want the race condition to happen here actually, we want to test that `dropgz` as an init container to `cns` for windows nodes, does not have any issues replacing the `azure-vnet.exe` binary during a pod scaling event (and it's good, no issues)